src/requirements/reqs-functions.ts

```diff
@@ -285,46 +214,135 @@ export function computeRequirements(
     reqs: computeUniversityRequirementFulfillments(coursesTaken)
   });
 
-  // PART 2: check college requirements
+  // PART 2: check college & major requirements
   if (!(college in requirementJson.college)) throw new Error('College not found.');
   const collegeReqs = requirementJson.college[college];
 
-  groups.push({
-    groupName: 'College',
-    specific: college,
-    reqs: computeCollegeOrMajorRequirementFulfillments(coursesTaken, collegeReqs.requirements)
+  type DecoratedCollegeOrMajorRequirementWithSourceType = DecoratedCollegeOrMajorRequirement & {
```

Comment:
Do we want to rename this to DecoratedRequirement or DecoratedCollegeMajorMinor Requirement, since it seems to handle every case now?